src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.java

```diff
@@ -49,46 +48,71 @@ public AbsentReasonsDto getAbsenceReasons() {
     public AttendanceDto createAttendance(final CreateAttendanceDto attendanceDto) {
 
         var attendance = attendanceRepository.save(toAttendance(attendanceDto));
-        publishAttendanceAndIEPWarning(attendance);
+
+        postNomisAttendance(attendance);
+        postIEPWarningIfRequired(
+                attendance.getOffenderBookingId(),
+                attendance.getCaseNoteId(),
+                attendance.getAbsentReason(),
+                attendance.getComments()
+        ).ifPresent(attendance::setCaseNoteId);
 
```

Comment:
Just to check I've understood this correctly - postIEPWarningIfRequired could return a CaseNote Id or it could return empty. If it does return a CaseNote Id, we set that on the attendance. 

Would id matter if we tried to set an empty case note id on the attendance?

Reply:
That's correct, however, if the case note id is empty nothing will happen. The ifPresent only executes if there is a value.

Reply:
Yeap, gotcha. Updating is quite an involved method now (and growing with my work), but I dont't have better ideas right now :D